### PR TITLE
Automatically import FIG data points from JSON

### PR DESCRIPTION
**Pull request by schbetsy**
Upon save, preview, or publish, a FIGContentPage now downloads JSON (following a specific structure representing FIG data points) from a specified URL, parses, and saves it to the page. The content editor places a FigDataPointsBlock in whatever Section of the page they want the data points to appear in.

On page load, the data point info is placed into the data points template in the section specified by the content editor. Links to the data points are added to the table of contents.

## Additions

- Data models (with migrations), an import script, a jinja template, to support the new ability to import structured JSON and have it inserted into a page

## Changes

- Minor changes to things like FIG section numbering to support the change

## How to test this PR

1. Running this branch, in Wagtail, create a Fig Content Page
2. In a Fig Section in the content of the page, add a Fig Data Points block
3. Find or create a structured JSON file of FIG data points. An example is at GHE/raw/CFPB/1071-fig-content/blob/master/test_data_points.json
4. In the Data Points tab of the Wagtail editor, enter the download URL of the JSON file
5. Preview or publish the page. 
6. See that the data points are displayed in basically the format we're going for
7. See that the table of contents correctly links to the data points and the anchor links are sensible

## Notes and todos

- ~~Ensure that the data points block is only inserted at the Section level, since that's the only way the formatting and section numbering make sense.~~
- Also, find a nicer way to validate the JSON schema. Consider using Wagtail Snippets to hold the JSON data, rather than downloading from a separate location.
- Document how to use this feature in Hubcap

## Checklist

- [x] PR has an informative and human-readable title
  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code follows the standards laid out in the [CFPB development guidelines](https://github.com/cfpb/development)
- [ ] Future todos are captured in comments and/or tickets
- [ ] Project documentation has been updated, potentially one or more of:
  - [This repo’s docs](https://cfpb.github.io/consumerfinance.gov/) (edit the files in the `/docs` folder) – for basic, close-to-the-code docs on working with this repo
  - CFGOV/platform wiki on GHE – for internal CFPB developer guidance
  - CFPB/hubcap wiki on GHE – for internal CFPB design and content guidance

### CODE REVIEW

File: cfgov/filing_instruction_guide/models.py
Comment (contolini) [inline cfgov/filing_instruction_guide/models.py:49]: When I pass it a [malformed file](https://GHE/raw/CFPB/1071-fig-content/master/test_data_points/malformed.json) (there's an extra curly brace on line 3) it raised this generic exception (`The file could not be downloaded at the specified URL.`) instead of a `JSONDecodeError`.

Is it possible to return some sort of `Your JSON is borked` message? I anticipate users will often provide invalid JSON because it's easy to include erroneous brackets and quotes.

File: cfgov/filing_instruction_guide/models.py
Comment (contolini) [inline cfgov/filing_instruction_guide/models.py:38]: Highlighting the missing key is super helpful here. 👍

**Review by contolini — CHANGES_REQUESTED** [review]
Nice job! I fed it all sorts of JSON files (see the `test_data_points/` dir in the `1071-fig-content` repo) and it handled them all gracefully, even the one with 1000 data points.

Couple questions below.

File: cfgov/filing_instruction_guide/models.py
Comment (schbetsy) [reply cfgov/filing_instruction_guide/models.py:49]: Good call. I was catching the JSON error incorrectly. Fixed in 485f794.

File: cfgov/filing_instruction_guide/jinja2/filing_instruction_guide/data_points.html
Comment (schbetsy) [reply cfgov/filing_instruction_guide/jinja2/filing_instruction_guide/data_points.html]: Turns out I don't know how jinja templates are supposed to be indented, but I decided to use this opportunity to totally rearrange that file and use macros to split it out. Done in 848f42179d2087ea079060f10df85491e427da92.

File: cfgov/filing_instruction_guide/blocks.py
Comment (schbetsy) [reply cfgov/filing_instruction_guide/blocks.py:47]: Good call. Done in 40fee9efa60abca6e3febdcdd1596d956d3376f3.

**Review by contolini — APPROVED** [review]

**Comment by schbetsy** [comment]
Thank you for reviewing, @contolini , and especially for testing with several different json files. I'm going to re-make all of the migrations before I merge, and double-check with designers to make sure it looks okay.

**Comment by contolini** [comment]
Cool I'll wait until this gets merged, rebase https://github.com/cfpb/consumerfinance.gov/pull/7268, and then deploy it to dev4 so we can have everything there to test at once. No rush.

**Pull request merged**
